plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/AddFileEntry.java

```diff
@@ -61,10 +61,35 @@ public AddFileEntry(
             @JsonProperty("parsedStats") Optional<DeltaLakeParquetFileStatistics> parsedStats,
             @JsonProperty("tags") @Nullable Map<String, String> tags,
             @JsonProperty("deletionVector") Optional<DeletionVectorEntry> deletionVector)
+    {
```

Comment:
> In case that there is checkpoint filtering applied
and there are partition constraints which do not
match the partition values of the entry, avoid
early to create the `AddFileEntry` instance.


Thanks for a detailed commit message, this is useful!

One thing you can add there is the _why_. Why is this change useful?
A naive reader may feel like we're saving an object construction (less GC pressure), while I guess the goal here is to prevent redundant  loads of the lazy blocks coming out of the Parquet reader and thus saving some resources (IO? CPU time for decoding?)